synthesizer/program/src/logic/command/get.rs

```diff
@@ -187,9 +288,21 @@ impl<N: Network> Display for Get<N> {
 
 impl<N: Network> FromBytes for Get<N> {
     /// Reads the command from a buffer.
-    fn read_le<R: Read>(mut reader: R) -> IoResult<Self> {
-        // Read the mapping name.
-        let mapping = CallOperator::read_le(&mut reader)?;
+    fn read_le<R: Read>(reader: R) -> IoResult<Self> {
+        // Peek at the first byte.
+        // TODO (howardwu): For mainnet - Read a `MappingLocator`.
+        let mut buffered = BufReader::new(reader);
+        let first_byte = {
+            let buffer = buffered.fill_buf()?;
+            buffer.first().copied()
+        };
+        let mut reader = buffered.into_inner();
+        // If the first byte is zero, then read a `MappingLocator`, otherwise read an `Identifier`.
+        let mapping = match first_byte {
+            Some(0u8) => MappingLocator::read_le(&mut reader)?,
+            Some(_) => MappingLocator::Resource(Identifier::read_le(&mut reader)?),
```

Comment:
What about the programs with the new serialization, where it is expecting the addition of the `version` and the `variant` bytes from `MappingLocator::Resource`?